### PR TITLE
Deduplicate and consolidate contributing information to community repo

### LINKED ISSUE #420: De-dupe developer/contributor information
@bgrant0607 recently noted that [Our Developer/Contributor Docs Need Love](https://groups.google.com/forum/#!topic/kubernetes-wg-contribex/ucUPGVKhkQI).

One aspect of that is removal of duplicated or outdated info.

To close this issue, in the [code repo](https://github.com/kubernetes/kubernetes), replace all long discussions of code contribution with shorter text pointing to the right spots in the [community repo](https://github.com/kubernetes/community).  As necessary, move or merge text to community.

### PR DESCRIPTION
**Pull request by monopole**
* Move remaining community stuff out of kubernetes/kubernetes.
  * Put self-contained things into their own MD files.
  * Point to those files rather than repeat their content.
  * Make all docs smaller, but more of them.

Companion to https://github.com/kubernetes/kubernetes/pull/42586

Closes #420

### CODE REVIEW

**Comment by monopole** [comment]
PTAL @grodrigues3

File: CLA.md [hunk 1]
```diff
@@ -1,30 +1,66 @@
...
+This policy does not apply to [third_party] and [vendor].
 
-**Step 1**: You must sign in with GitHub.
+## Where do I sign?
 
-**Step 2**: If you are signing up as an employee, you must use your official person@organization.domain email address in the CNCF account registration page.
+Review the [CLA for individuals] and visit the [individual signup form] to sign
+as an individual or as an employee of a signed organization.
 
+To sign as an organization, read the [CLA for corporations] and visit [corp signup].
```
Comment (grodrigues3) [inline CLA.md]: Nit: I think the link to the sign-up page should be in the "How do I sign section"

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,16 +1,16 @@
...
-The process is as follows:
+Contributions to this community repository follow a
+[pull request](https://help.github.com/articles/using-pull-requests/)
+model with some additional process:
 
 1. Submit a pull request with the requested change.
-2. Another person, other than a Special Interest Group (SIG) owner, can mark it Looks Good To Me (LGTM) upon successful review. Otherwise feedback can be given.
-3. A SIG owner can merge someone else's change into their SIG documentation immediate.
-4. Someone cannot immediately merge their own change. To merge your own change wait 24 hours during the week or 72 hours over a weekend. This allows others the opportunity to review a change.
+2. Another person, other than a [SIG lead](sig-list.md), can mark it Looks Good To Me (LGTM) upon successful review.
```
Comment (grodrigues3) [inline CONTRIBUTING.md]: Can we reorganize this so that it's clear that you must do either (2+4) or 3.

File: communication.md [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+The Kubernetes community abides by the [CNCF code of conduct].  Here is an excerpt:
+
+> _As contributors and maintainers of this project, and in the interest
+> of fostering an open and welcoming community, we pledge to respect
+> all people who contribute through reporting issues, posting feature
+> requests, updating documentation, submitting pull requests or patches,
+> and other activities._
+
+## SIGS
```
Comment (grodrigues3) [inline communication.md]: Nit: lowercase s at the end to be consistent with usage below.

**Comment by grodrigues3** [comment]
Generally LGTM minus the few nits.  Thanks so much for doing this!

File: CLA.md (hunk 1 above)
Comment (monopole) [inline CLA.md]: Done.   Just got rid of the (very short) "Where do I sign" section.

File: CONTRIBUTING.md (hunk 2 above)
Comment (monopole) [inline CONTRIBUTING.md]: Attempted reorg.

File: communication.md (hunk 3 above)
Comment (monopole) [inline communication.md]: Done

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -1,16 +1,30 @@
...
+#### 1. Submit a PR with your change
 
-1. Submit a pull request with the requested change.
-2. Another person, other than a Special Interest Group (SIG) owner, can mark it Looks Good To Me (LGTM) upon successful review. Otherwise feedback can be given.
-3. A SIG owner can merge someone else's change into their SIG documentation immediate.
-4. Someone cannot immediately merge their own change. To merge your own change wait 24 hours during the week or 72 hours over a weekend. This allows others the opportunity to review a change.
+#### 2. Get an LGTM.
 
-_Note, the SIG Owners decide on the layout for their own sub-directory structure._
+Upon successful review, someone will give the PR
```
Comment (ravisantoshgudimetla) [inline CONTRIBUTING.md]: "someone from reviewers/approvers in owners list file" perhaps if its not too verbose?

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -1,16 +1,30 @@
...
+a __LGTM__ (_looks good to me_) in the review thread.
+   
+#### 3. Allow time for others to see it
+
+Once you have an __LGTM__, please wait 24 hours during
+the week or 72 hours over a weekend before you
+merge it, to give others (besides your initial reviewer) 
+time to see it.
+
```
Comment (ravisantoshgudimetla) [inline CONTRIBUTING.md]: "before you merge it" to "before it gets merged" perhaps, as most of developers won't have write access to repo.

**Pull request merged**